src/Input/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -1,68 +1,55 @@
 import * as React from 'react';
+import classNames from 'classnames';
 
-import { Icon, SecondaryColor } from '../../index';
-import { Toggle, ToggleBall, ToggleSwitchWrapper } from './Styles';
+import { SecondaryColor } from '../../Style/Colors';
+import Icon from '../../General/Icon';
+import { Toggle, ToggleBall } from './Styles';
 
 export interface ToggleSwitchProps {
   defaultActive?: boolean;
-  activeBackgroundColor?: string;
-  inactiveBackgroundColor?: string;
-  activeBallColor?: string;
-  inactiveBallColor?: string;
-  activeIconName?: string;
-  inactiveIconName?: string;
+  iconOptions?: {
+    active: string;
+    inactive: string;
+  };
   onChange?: (active: boolean) => void;
-  handleClick: (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => void;
+  className?: string;
 }
 
 const ToggleSwitch: React.FunctionComponent<ToggleSwitchProps> = ({
   defaultActive,
-  activeIconName,
-  inactiveIconName,
-  activeBackgroundColor,
-  inactiveBackgroundColor,
-  activeBallColor,
-  inactiveBallColor,
+  iconOptions,
   onChange,
+  className,
+  ...defaultProps
 }) => {
   const [active, setActive] = React.useState(defaultActive);
 
   return (
-    <ToggleSwitchWrapper>
-      <Toggle
-        aria-checked={active}
-        role="switch"
-        onClick={() => {
-          const newActiveState = !active;
-          setActive(newActiveState);
-          if (onChange) onChange(newActiveState);
-        }}
-        active={active}
-        activeBackgroundColor={activeBackgroundColor}
-        inactiveBackgroundColor={inactiveBackgroundColor}
-      >
-        <ToggleBall
-          active={active}
-          activeBallColor={activeBallColor}
-          inactiveBallColor={inactiveBallColor}
-        >
-          {activeIconName && inactiveIconName && (
-            <Icon
-              name={active ? activeIconName : inactiveIconName}
-              color={active ? activeBackgroundColor : inactiveBackgroundColor}
-            />
-          )}
-        </ToggleBall>
-      </Toggle>
-    </ToggleSwitchWrapper>
+    <Toggle
+      aria-checked={active}
+      role="switch"
+      className={classNames('aries-toggle-switch', className)}
+      onClick={() => {
+        const newActiveState = !active;
+        setActive(newActiveState);
+        if (onChange) onChange(newActiveState);
+      }}
+      active={active}
+      {...defaultProps}
+    >
+      <ToggleBall active={active}>
+        {iconOptions && (
```

Comment:
This will error if either `active` or `inactive` properties are not present, or an empty object is passed. Can use lodash `has` or `hasOwnProperty` to check for both properties.